### PR TITLE
Renamespace & fix build

### PR DESCRIPTION
**Pull request by machour**
- Changed namespace from \yii\cache to \Yiisoft\Cache
- Removed init() methods (still needs some fixing)
- FileCache doesn't resolve aliases anymore
- Fixed tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -85,8 +85,8 @@ install:
   # setup PHP extension
   - |
     if [[ $TASK_TESTS_PHP == 1 && $TRAVIS_PHP_VERSION != nightly ]]; then
-      vendor/yiisoft/yii-core/tests/data/travis/apcu-setup.sh
-      vendor/yiisoft/yii-core/tests/data/travis/memcache-setup.sh
+      tests/travis/apcu-setup.sh
```
Comment (machour) [inline .travis.yml]: apcu setup compilation is failing on travis/7.3

File: src/ApcCache.php [hunk 2]
```diff
@@ -35,47 +33,31 @@
...
 {
-    /**
-     * Initializes this application component.
-     * It checks if extension required is loaded.
-     * @throws \yii\exceptions\InvalidConfigException
-     */
-    public function init()
-    {
-        parent::init();
-        if (!\extension_loaded('apcu')) {
```
Comment (machour) [inline src/ApcCache.php]: `ext.apcu` will be added as a requirement in the driver's composer.json instead

File: src/MemCached.php [hunk 3]
```diff
@@ -95,40 +92,43 @@ class MemCached extends SimpleCache
...
-        $this->addServers($this->getMemcached(), $this->getServers());
+        parent::__construct($serializer);
+
+        if (empty($servers)) {
+            $servers = [new MemCachedServer('127.0.0.1')];
+        }
+
+        $this->_servers = $servers;
+
+        $this->addServers($this->getMemcached(), $this->_servers);
```
Comment (machour) [inline src/MemCached.php]: Not good as the $options, $username, $password properties are still not set.

Same old question: how do we manage a constructor that should take a lot of parameters without Yii::buildObject() .. Is it worth having yiisoft/objects with some objects goodies?

Comment (samdark) [reply src/MemCached.php]: Is that really a lot of parameters?

File: src/FileCache.php [hunk 4]
```diff
@@ -93,7 +90,7 @@ public function __construct(string $cachePath = '@runtime/cache', $serializer =
      */
     public function setCachePath(string $cachePath)
     {
-        $this->cachePath = Yii::getAlias($cachePath);
+        $this->cachePath = $cachePath;
```
Comment (machour) [inline src/FileCache.php]: Dropping alias resolving here, as it should be done via configuration instead for Yii apps, correct?

File: src/MemCached.php (hunk 3 above)
Comment (machour) [reply src/MemCached.php]: 4 properties in MemCached:
  * $persistentId
  * $options
  * $username
  * $password

2 properties at SimpleCache level:
  * $defaultTtl
  * $keyPrefix

and $serializer & $servers that are already in constructor.

It's a lot.

Comment (samdark) [reply src/MemCached.php]: Does not look too bad to me...

File: src/FileCache.php (hunk 4 above)
Comment (samdark) [reply src/FileCache.php]: Yes.

File: src/MemCached.php (hunk 3 above)
Comment (machour) [reply src/MemCached.php]: Let's deal with this later.

If this PR looks ok to you as is, I'll merge it tomorrow and take care of updating references.

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**